### ISSUE TITLE
Decode and play DTS audio in FLAC file

### ISSUE THREAD

**Issue by renedrive** [fooyin/fooyin#346]
It seems that in Linux there is no player that can play FLAC files that contain DTS multi-channel audio format.
I have many of these files, some of which purchased from well-known HiFi HiEnd sites.
Typically I use VLC as a player but even it does not play them correctly, it only plays digital noise, and Fooyin seems to do the same.
The only solution to listen to these files that I have found is to use Foobar2000 through Wine, which is not the best solution, but it works very well.
It would be nice to be able to listen to these FLAC files with DTS multi-channel audio in Fooyin! ;-)
Thanks for your attention, and I wish you good work.
Renato, from Italy

PS: FLAC files with DTS multi-channel audio encoded are easy to find but I remain at your disposal, if you can't find any, to pass you some excellent ones (e.g. Spyro Gyra Fusion)

**Comment by diizzyy** [comment]
What does ffmpeg say about such a file?
```ffmpeg -i filename.flac``` should do it

**Comment by Vendrell** [comment]
> Typically I use VLC as a player but even it does not play them correctly, it only plays digital noise, and Fooyin seems to do the same.

VLC seems to decode my DTS wav files properly, I mean it actually plays them. Fooyin, Strawberry output the noise.
Fedora 43.

**Comment by renedrive** [comment]
You're absolutely right, I also play DTS WAV files correctly in both VLC and Fooyin.
As mentioned, however, files in the FLAC container don't play correctly in either VLC or Fooyin.
To play DTS files correctly in the FLAC container, I have to use Foobar2000 with Wine.
I haven't found a player in the Linux world that correctly plays DTS files in the FLAC container, which is why I have to use Foobar2000.
As mentioned, music files in the FLAC container with DTS encoding aren't unknown or unusual; they're available for sale in high-end online music stores, so they're considered a current standard.

**Comment by Vendrell** [comment]
Looks like Elisa is also able to play my files just fine, just checked. Fooyin can't handle Dynamic Range 11 DTS .wav on my machine.

**Comment by renedrive** [comment]
You're right, I agree. Elisa has no problem playing DTS-encoded WAV files.
Many Linux players are able to play DTS-encoded files correctly in WAV containers, though not all.
But, as mentioned, none seem to be able to play them by decoding DTS if the container is FLAC.
At least not among the ones I've tried, and Fooyin is one of them.

**Comment by alekksander** [comment]
do you guys have a sample of such file to play around with?

**Comment by Vendrell** [comment]
> do you guys have a sample of such file to play around with?

I've found a smaller 14 seconds track (2.4mb). Where do you want it?

**Comment by renedrive** [comment]
I own many titles encoded in DTS and compressed with FLAC.
Like these:
Spyro Gyra - 2001 - In Modern Times (DTS)
Spyro Gyra - 2007 - Good To Go-Go (DTS)
Spyro Gyra - 2004 - The Deep End (DTS)
Miles Davis - 1963 - Seven Step To Heaven (DTS 5.1)
Ray Brown - 1998 - Summertime (Trio with Ulf Wakenius) (DTS)
Norah Jones - 2004 - Live In 2004 (DTS)
Diana Krall - 1997 - Love Scenes (DTS)
The Crystal Method - 2004 - Legion Of Boom [DTS]
Jean Michel Jarre - 2003 - Geometry Of Love (DTS)
Jean Michel Jarre - 2021 - Welcome To The Other Side [DTS]
Dire Straits - 1982 - Love Over Gold (DTS)
But there are only a couple 5.1, most are stereo.
There are many others for sale; for example, I even have a Santana one somewhere, but I can't find it now.
However, none of them are read and decoded correctly by Linux player software, not even the stereo ones. (Not VLC, Haruna, MPlayer, SMPlayer, MPV, Kaffeine, fooyin, etc.)
I'm sharing a Miles Davis track on the Mega Cloud storage system. Here's the link.
[https://mega.nz/file/LLxi3ZxS#Y00V_Ar5y7WllCS43VeamdKrPb2QYQiKaxn4-OE4qPo](url)
And this is a 5.1-channel track.
If you also want a 2-channel one, this one by Diana Krall is the right one.
Here's the link.
[https://mega.nz/file/HCxDEDjC#ZcOmGwnh1285gPsBcjpa3MV4R8Yc4JawyQjkl2_Fp4U](url)
I hope they're useful.

**Comment by alekksander** [comment]
@renedrive 
miles davis file works in fooyin (even) in stereo once proper output is selected. could probably work with ALSA→ hw:x,x as well, but needs dsp→downmix to do that i guess. ffmpeg is able to decode it properly.
krall file is something else… dts-cd hidden in flac container… it's new to me to be honest, but internet explains back in the day it was the only way to digitalize dts discs. fooyin probably needs a plugin for that, just like foobar does – because ffmpeg cannot play it without some tricks. 

therefore, if i may – i suggest donating to the project and be patient until dev has time and processing power to bless us with proper dts plugin :)

**Comment by renedrive** [comment]
Thanks for the info.
I admit that the tests I ran were a while ago; I hadn't retried all the files recently; I only had the 2-channel ones on hand, so I didn't notice that the 5.1 ones started working.
You're right, Foojin plays 5.1-channel files without any problems, and now I've verified that VLC has also started working for those files.
I can confirm that the 2-channel ones, in stereo, still don't work.

I filed the same report with the VLC and ffmpeg teams some time ago, when even the 5.1-channel files weren't working.
I made this one a while later.
But I agree, a donation and a request for implementation can't hurt.
Thanks for the info.

It was actually a transitional format, before DVD Audio or SACD, and then the recent HD files. It doesn't make much sense today, but those of us who collected them still play them.
It also made sense with a 5.1-channel format, but less so with a 2-channel format, given that DTS is a lossy compressed format, and even though it had higher definition, it still had compression losses compared to a good PCM CD.
It was probably more marketing than anything else, to have a "different" format to sell in the HiFi Hi-End market.
